### PR TITLE
Added WebAssembly to the WorkerDOM GlobalScope

### PR DESCRIPTION
**Pull request by torch2424**
This adds the [WebAssembly](https://developer.mozilla.org/en-US/docs/WebAssembly) Object to the WorkerDOM GlobalScope. Thus, allowing WebAssembly to be used within the worker by users 😄 🎉

This also includes a demo built with [AssemblyScript](https://github.com/AssemblyScript/assemblyscript), and is a simple Wasm equivalent of a hello world, that adds `i32` values. 

# Example

wasmWorkerDom

### CODE REVIEW

File: demo/webassembly/index.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
```
Comment (torch2424) [inline demo/webassembly/index.html]: From @kristoferbaxter Make sure this demo matches the updated structure

File: src/worker-thread/WorkerDOMGlobalScope.ts [hunk 2]
```diff
@@ -53,6 +53,7 @@ import { HTMLElement } from './dom/HTMLElement';
 export interface GlobalScope {
   initialize: (document: Document, strings: Array<string>, hydrateableNode: HydrateableNode, keys: Array<string>) => void;
   navigator: WorkerNavigator;
+  WebAssembly: object;
```
Comment (torch2424) [inline src/worker-thread/WorkerDOMGlobalScope.ts]: Leave a TODO, should we type this more explicitly?

File: src/worker-thread/index.amp.ts [hunk 3]
```diff
@@ -133,6 +134,7 @@ const WHITELISTED_GLOBALS = [
 
 const globalScope: GlobalScope = {
   navigator: (self as WorkerGlobalScope).navigator,
+  WebAssembly: (self as any).WebAssembly,
```
Comment (torch2424) [inline src/worker-thread/index.amp.ts]: `self as WorkerGlobalScope`

File: src/worker-thread/index.ts [hunk 4]
```diff
@@ -52,6 +52,7 @@ import { MutationObserver } from './MutationObserver';
 
 const globalScope: GlobalScope = {
   navigator: (self as WorkerGlobalScope).navigator,
+  WebAssembly: (self as any).WebAssembly,
```
Comment (torch2424) [inline src/worker-thread/index.ts]: `self as WorkerGlobalScope`

File: demo/webassembly/webassembly.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const wasmModuleUrl = 'http://localhost:3001/webassembly/webassembly.wasm';
+const importObject = {};
+const wasmBrowserInstantiate = async () => {
+  let response = undefined;
+
+  if (WebAssembly.instantiateStreaming) {
```
Comment (torch2424) [inline demo/webassembly/webassembly.js]: From @kristoferbaxter, let's comment this functionality for future readers.

**Comment by torch2424** [comment]
TODO:

* Update the PR
* Make sure it works
* Write a better description
* Kris will merge, approve, and release a new minor

**Comment by torch2424** [comment]
Also, I'll go ahead and add a `README.md` to the demo, to give instructions on how to build the AssemblyScript file 😄

File: demo/webassembly/index.html (hunk 1 above)
Comment (torch2424) [reply demo/webassembly/index.html]: This is the same as the hello-world, so done 😄

**Comment by torch2424** [comment]
So I addressed all comments, and this is good to go! @kristoferbaxter 😄 

Only one thing, I tried to do the: `self as WorkerGlobalScope`, but I don't think WebAssembly is provided on the `WorkerGlobalScope`?

Screen Shot 2019-06-20 at 6 23 52 PM

Though, I found `WebAssembly` and `OffscreenCanvas` on: https://github.com/microsoft/TypeScript/blob/master/src/lib/webworker.generated.d.ts

Let me know what you think! Thanks! 👍

**Comment by torch2424** [comment]
@kristoferbaxter Woo! Thank you! 😄 🎉

**Comment by kristoferbaxter** [comment]
Thank you @torch2424! This is a great PR.

**Pull request merged**